LightGBM/data.py

```diff
@@ -91,13 +101,46 @@ def optimize_df(calendar, prices, sales, days=None, verbose=False):
     return calendar, prices, sales
 
 
+def melt_and_merge(calendar, prices, sales, submission=False):
+    """
+    Convert sales from wide to long format, and merge sales with
+    calendar and prices to create one dataframe.
+
+    Args:
+        calendar = [pd.DataFrame] dates of product sales
+        prices   = [pd.DataFrame] price of the products sold per store and date
+        sales    = [pd.DataFrame] historical daily unit sales data per product and store 
+
+    Returns [pd.DataFrame]:
+        Merged long format dataframe
+    """
+    id_cols = ['id', 'item_id', 'dept_id','store_id', 'cat_id', 'state_id']
+    if submission:
+        sales.drop(sales.columns[6:-MAX_LAG], axis=1, inplace=True)
+        for day in range(LAST_DAY + 1, LAST_DAY + 28 + 1):
+            sales[f"d_{day}"] = np.nan
+
+    df = pd.melt(sales,
```

Comment:
Keyword arguments usually do not have spaces around the equals sign. PyCharm should have warned you about this.

Reply:
I think you looked at an older commit (the spaces were there because it was copy-pasted from the kaggle notebook), but it should be good in the later commits. I use VS Code instead of PyCharm so I don't think I get these warnings.